docs/guides/4-custom-rulesets.md

```diff
@@ -265,3 +269,112 @@ Several functions [are provided by default](../reference/functions.md) for your
 ## Custom Functions
 
 If none of the [core functions](../reference/functions.md) do what you want, you can [write your own custom functions](./5-custom-functions.md).
+
+## Alternative JS Ruleset Format
+
+Spectral now supports an alternative format to write rulesets in Javascript using a similar syntax.
+
+### Benifits
+
+- Explicitly load formats or rulesets to get control over versioning.
+- Load common functions from popular JS libraries easily
+
+### Example
+
+```js
+//you can import popular functions from libraries
+import { isObject } from "https://cdn.jsdelivr.net/npm/lodash-es/+esm";
+import { truthy, schema } from "https://cdn.jsdelivr.net/npm/@stoplight/spectral-functions/+esm";
+// you can stick to an older version if you want to for some reason. That's fine
+import { alphabetical } from "https://cdn.jsdelivr.net/npm/@stoplight/spectral-functions@1.0.4/+esm";
+import { oasRuleset } from "https://cdn.jsdelivr.net/npm/@stoplight/spectral-rulesets/+esm";
+import { oas2 } from "https://cdn.jsdelivr.net/npm/@stoplight/spectral-formats/+esm";
+
+import { verifyType } from "./verifyType.mjs";
+
+const $SCHEMA_DRAFT_2020_XX_REGEX = /^https?:\/\/json-schema.org\/draft\/2020-\d\d\/(?:hyper-)?schema#?$/;
+
+const JSONSchemaDraft2020_XX = document => isObject(document) && "$schema" in document && $SCHEMA_DRAFT_2020_XX_REGEX.test(document.$schema);
+
+export default {
+  formats: [oas2, oas3],
+  // No longer accepts functions & functionOptions
+  extends: [oasRuleset],
+  rules: {
+    "valid-rule": {
+      given: "$.info",
+      then: {
+        function: truthy, // instead of 'truthy'
+      },
+    },
+    "only-new-json-schema": {
+      formats: [JSONSchemaDraft2020_XX],
+      given: "$..type",
+      then: {
+        function: verifyType,
+      },
+    },
+  },
+};
+```
+
+## Aliases
+
+Targeting certain parts of an OpenAPI spec is powerful but it can become cumbersome to write and repeat complex JSON path expressions across various rules. Define aliases for commonly used JSON paths on a global level which can then be reused across the ruleset.
+
+Aliases can be defined in an array of KVP on top of the ruleset.
+
+```
+aliases:{
+  {Name}: '{JSONPath}'
+}
+```
+
+### Example:
+
+```json
+{
+  "aliases": {
+    "Info": "$..info",
+    "InfoDescription": "#Info.description",
+    "InfoContact": "#Info.contact",
+    "InfoContactName": "#InfoContact.name"
+  }
+}
+```
+
+> This will be followed by our core rulesets providing a common set of aliases for OpenAPI and AsyncAPI so that our users don't have to do the work at all. If you have ideas about what kind of aliases could be useful leave your thoughts [here](https://roadmap.stoplight.io).
+
+## Overrides
```

Comment:
we used to have exceptions in a seprate file - not saying we have to do it for overrides, but it's okay to keep them separately if you want to.